relayer/chain/substrate/writer.go

```diff
@@ -85,16 +90,32 @@ func (wr *Writer) write(c types.Call) error {
 		return err
 	}
 
-	var accountInfo types.AccountInfo
-	ok, err := wr.conn.api.RPC.State.GetStorageLatest(key, &accountInfo)
-	if err != nil {
-		return err
-	}
-	if !ok {
-		return fmt.Errorf("no account info found for %s", wr.conn.kp.URI)
-	}
+	// Poll account state until we have a new nonce. This matches our extrinsic
+	// sending rate with the node's extrinsic processing rate, and avoids
+	// the transaction being dropped from the node's transaction pool.
```

Comment:
It limits the intake of the substrate chain though to 1 extrinsic per block. Is that really what we want? There might be severe delays if for some reason we need to catch up after a downtime. 

A less restrictive way here would be to have a sending queue and just back off exponentially if we hit errors.

Any thoughts @musnit ?

Reply:
i think 1 extrinsic per block should be fine - remember substrate is 10 - 100x faster than ethereum, especially if we end up submitting transaction proofs in the same extrinsic as new block header. lets not prematurely optimize :), though could create a github issue for it

Reply:
@philipstanislaus this is not intended to be a final implementation. It's a workaround for dropped transactions (see https://stackoverflow.com/questions/63475813/transaction-failed-error-1014-priority-is-too-low-515000139-vs-515000139). Only one transaction with `(sender, nonce)` is allowed in the pool at a time. But I don't know much about this stuff in Substrate. Does the nonce here have to match the `accountInfo.Nonce`? If it can be an arbitrary number, I can easily fix this. @vgeddes maybe you know?

Reply:
There's also the possibility of customizing the transaction pool logic in our Substrate node. But I want to avoid doing that right now

Reply:
oh yeh there could be some interesting designs that come out when considering node transaction pool changes too - but yeh lets not worry about for now

Reply:
Created an issue for this https://github.com/Snowfork/polkadot-ethereum/issues/170

Reply:
Sounds good to me. Nonce needs to be sequential, if we don't customize the pool.